src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,177 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Represents the component of the Duke program
+ * that makes sense of user input.
+ */
+public class Parser {
+
+    protected static final DateTimeFormatter INPUT_DATETIME_FORMAT = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+
+    /**
+     * Returns a valid Duke Command after parsing user input.
+     *
+     * @param input user input.
+     * @return a valid Duke Command
+     * @throws DukeException when user input
+     *                       does not match any valid Duke Commands.
+     */
+    public Command parseCommand(String input) throws DukeException {
+        String[] arr = input.split(" ");
+        try {
+            return Command.valueOf(arr[0].toUpperCase());
+        } catch (IllegalArgumentException illegalArgumentException) {
+            throw new DukeException("OOPS!!! I'm sorry, but that is an invalid command :-(");
+        }
+    }
+
+    /**
+     * Returns a ToDo based on user input.
+     *
+     * @param input user input.
+     * @return a ToDo based on user input.
+     * @throws DukeException when user input does not contain
+     *                       the description of the ToDo after the command.
+     */
+    public ToDo parseToDo(String input) throws DukeException {
+        String[] arr = input.split(" ");
+        if (arr.length < 2) {
+            throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+        }
+        return new ToDo(input.substring(5));
+    }
+
+    /**
+     * Returns a Deadline based on user input.
+     *
+     * @param input user input.
+     * @return a Deadline based on user input.
+     * @throws DukeException when user input does not contain
+     *                       the description of the Deadline after the command,
+     *                       the date and time of the Deadline after '/by'
+     *                       or when the date and time is not in the correct format.
+     */
+    public Deadline parseDeadline(String input) throws DukeException {
+        String[] arr = input.split(" ");
+        String[] arr2 = input.split(" /by ");
+        if (arr.length < 2 || arr2[0].split(" ").length < 2) {
+            throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+        }
+        if (arr2.length < 2) {
+            throw new DukeException("OOPS!!! You did not set a deadline using '/by'.");
+        }
```

Comment:
I think this one can be done better as both the if-blocks have common conditionns. The first condition has an OR condition so maybe that can be used to "differentiate" the exceptions thrown.